packages/hadron-app-registry/src/register-plugin.tsx

```diff
@@ -0,0 +1,200 @@
+import React, { useRef, useState } from 'react';
+import type { Store as RefluxStore } from 'reflux';
+import { Provider as ReduxStoreProvider } from 'react-redux';
+import type { Actions } from './actions';
+import { type Store, type AppRegistry, isReduxStore } from './app-registry';
+import { useGlobalAppRegistry, useLocalAppRegistry } from './react-context';
+
+function LegacyRefluxProvider({
```

Comment:
BTW, this is something we can use as a complete replacement for `packages/compass-components/src/components/store-connector.tsx` allowing us to remove reflux dependency from compass-components (but also can keep as is for now until we just clean it up by setting up plugins using this new interface)